acnportal/algorithms/tests/generate_test_cases.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+
+
+def session_generator(
+    num_sessions,
+    arrivals,
+    departures,
+    requested_energy,
+    remaining_energy,
+    max_rates,
+    min_rates=None,
+    station_ids=None,
+):
+    sessions = []
+    for i in range(num_sessions):
+        station_id = station_ids[i] if station_ids is not None else f"{i}"
+        session_id = f"{i}"
+        min_rate = min_rates[i] if min_rates is not None else 0
+        s = {
```

Comment:
I think this was mainly to avoid rewriting TestingInterface. However, with the new interface design you have been working on, maybe we can just return SessionInfo and InfrastructureInfo objects then pass them into the interface.